ipn/ipnstate/ipnstate.go

```diff
@@ -199,6 +199,10 @@ type PeerStatus struct {
 	OS      string // HostInfo.OS
 	UserID  tailcfg.UserID
 
+	// AltSharerUserID is the user who shared this node
+	// if it's different than UserID. Otherwise it's zero.
+	AltSharerUserID tailcfg.UserID `json:",omitempty"`
```

Comment:
Why Alt? Why not just SharerUserID

Reply:
Because the name "SharerUserID" (which I started with) is super misleading however it's documented. It makes it sound like it's always non-zero if the node was shared, which it's not.

The whole zero-means-same-as-UserID thing is a network optimization at best but sucks as an API.

Because PeerState is more of the API that end users will see (since tailcfg.Node is increasingly internal, as we remove it from IPNBus), I wanted to be super explicit in its name, crying out that it's weird.

Reply:
If we knew for sure the node was shared to us, then I could always set SharerUserID (and name it as such) but there's no bit we have on the wire that says that the node was shared.